sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/util/DataLakeSasImplUtil.java

```diff
@@ -0,0 +1,342 @@
+     *
+     * @param delegationKey {@link UserDelegationKey}
+     * @param accountName The account name
+     * @return A String representing the Sas
+     */
+    public String generateUserDelegationSas(UserDelegationKey delegationKey, String accountName) {
+        StorageImplUtils.assertNotNull("delegationKey", delegationKey);
+        StorageImplUtils.assertNotNull("accountName", accountName);
+
+        ensureState();
+
+        // Signature is generated on the un-url-encoded values.
+        final String canonicalName = getCanonicalName(accountName);
+        String signature = StorageImplUtils.computeHMac256(
+            delegationKey.getValue(), stringToSign(delegationKey, canonicalName));
+
+        return encode(delegationKey, signature);
+    }
+
+    /**
+     * Encodes a Sas from the values in this type.
+     * @param userDelegationKey {@link UserDelegationKey}
+     * @param signature The signature of the Sas.
+     * @return A String representing the Sas.
+     */
+    private String encode(UserDelegationKey userDelegationKey, String signature) {
+        /*
+         We should be url-encoding each key and each value, but because we know all the keys and values will encode to
+         themselves, we cheat except for the signature value.
+         */
+        StringBuilder sb = new StringBuilder();
+
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SERVICE_VERSION, version);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_PROTOCOL, this.protocol);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_START_TIME, formatQueryParameterDate(this.startTime));
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_EXPIRY_TIME, formatQueryParameterDate(this.expiryTime));
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_IP_RANGE, this.sasIpRange);
+        tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_IDENTIFIER, this.identifier);
+
+        if (userDelegationKey != null) {
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_OBJECT_ID,
+                userDelegationKey.getSignedObjectId());
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_TENANT_ID,
+                userDelegationKey.getSignedTenantId());
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_START,
+                formatQueryParameterDate(userDelegationKey.getSignedStart()));
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_EXPIRY,
+                formatQueryParameterDate(userDelegationKey.getSignedExpiry()));
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_SERVICE,
+                userDelegationKey.getSignedService());
+            tryAppendQueryParameter(sb, Constants.UrlConstants.SAS_SIGNED_KEY_VERSION,
+                userDelegationKey.getSignedVersion());
+
+            /* Only parameters relevant for user delegation SAS. */
```

Comment:
Is this comment not true for the whole if block?

Reply:
Yeah I wanted to more be like these are the only params relevant to user delegation sas but not actually tied to the user delegation key